### PR TITLE
Visible cooldown timer

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Ability type actions now display their actual cooldown time remaining instead of a random clock symbol.

https://github.com/tgstation/TerraGov-Marine-Corps/assets/7869430/cf7e665c-7db2-4b83-86c9-0ba0f023eeef
## Why It's Good For The Game
Useful information on your screen is good.
## Changelog
:cl:
qol: Ability type actions now display their cooldown remaining
/:cl:

### CODE REVIEW

File: code/game/objects/effects/countdown.dm [hunk 1]
```diff
@@ -88,3 +88,47 @@
...
+	name = "cooldown"
+	color = "#d1d1d1"
+	invisibility = SEE_INVISIBLE_LIVING
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	layer = ABOVE_HUD_LAYER
+	plane = HUD_PLANE
+	pixel_x = 5
+	pixel_y = 8
+	appearance_flags = KEEP_APART|RESET_COLOR
+	var/datum/action/ability/attached_action
```
Comment (ivanmixo) [inline code/game/objects/effects/countdown.dm:102]: Autodoc

File: code/game/objects/effects/countdown.dm [hunk 2]
```diff
@@ -88,3 +88,47 @@
...
+
+/obj/effect/countdown/action_cooldown/Destroy()
+	attached_action = null
+	return ..()
+
+/obj/effect/countdown/action_cooldown/attach(atom/A)
+	var/atom/movable/screen/action_button/button = A
+	if(istype(button))
+		return qdel(src)
```
Comment (ivanmixo) [inline code/game/objects/effects/countdown.dm:111]: Returning qdel? Any particular reason?

File: code/game/objects/effects/countdown.dm [hunk 3]
```diff
@@ -88,3 +88,47 @@
...
+	attached_action = button.source_action
+
+/obj/effect/countdown/action_cooldown/start()
+	if(!started)
+		START_PROCESSING(SSfastprocess, src)
+		started = TRUE
+
+/obj/effect/countdown/action_cooldown/process()
+	if(QDELETED(attached_to))
+		qdel(src)
```
Comment (ivanmixo) [inline code/game/objects/effects/countdown.dm:123]: Think you forgot a return below the qdel

**Review by ivanmixo — CHANGES_REQUESTED** [review]
Also make sure this works fine with the savage cooldown maptext on runner

**Comment by Lumipharon** [comment]
Yep, works fine with savage, that timer is positioned differently.
That should also probs be a countdown instead of an MA overlay though.

File: code/game/objects/effects/countdown.dm (hunk 2 above)
Comment (Lumipharon) [reply code/game/objects/effects/countdown.dm]: updated

**Review by ivanmixo — APPROVED** [review]

**Pull request merged**